CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -0,0 +1,172 @@
+    /// The possible roles that a player can have in Texas Hold Em.
+    /// The Small Blind is always left of the Dealer.
+    /// </summary>
+    public enum PlayerRole
+    {
+        DEALER,
+        SMALL_BLIND,
+        BIG_BLIND,
+    }
+    
+    public class TexasHoldEmManager
+    {
+        /// <summary>
+        /// The cardManager for the game.
+        /// </summary>
+        private CardManager _cardManager;
+        
+        /// <summary>
+        /// The user's hand of cards.
+        /// </summary>
+        private UserHand _userHand;
+        
+        /// <summary>
+        /// Flag that indicates if the user is still active in the game.
+        /// </summary>
+        private bool _gameOver;
+        
+        /// <summary>
+        /// The chip manager for the game.
+        /// </summary>
+        private ChipManager _chipManager;
+        
+        /// <summary>
+        /// The main pot for the game.
+        /// </summary>
+        private int _mainPot;
+        
+        /// <summary>
+        /// The side pot for the game.
+        /// </summary>
+        private int _sidePot;
+        
+        /// <summary>
+        /// The current bet for the game.
+        /// This is the amount that all players must match to stay in the game.
+        /// </summary>
+        private int _currentBet;
+        
+        /// <summary>
+        /// The Small Blind for the game.
+        /// </summary>
+        private int _smallBlind;
+        
+        /// <summary>
+        /// The Big Blind for the game.
+        /// </summary>
+        private int _bigBlind;
+        
+        /// <summary>
+        /// The Ante for the game.
+        /// This is the amount that all players must pay to play the game.
+        /// </summary>
+        private int _ante;
+        
+        /// <summary>
+        /// A dictionary that tracks which player has what role.
+        /// </summary>
+        private Dictionary<PlayerRole, int> _players { get; set; }
+        
+        private List<CardHand> _playerHands { get; set; }
+
+        public TexasHoldEmManager()
+        {
+            _cardManager = new CardManager();
+            _userHand = new UserHand();
+            _gameOver = false;
+            _chipManager = new ChipManager();
+            _mainPot = 0;
+            _sidePot = 0;
+            _currentBet = 0;
+            _smallBlind = 0;
+            _bigBlind = 0;
+            _ante = 0;
+
+            // Initialize the player hands. The first player is the user.
+            for (int i = 0; i < Constants.AI_PLAYER_COUNT; i++)
+            {
+                if (i == 0)
+                {
+                    _playerHands[i] = new UserHand();
+                }
+                else
+                {
+                    _playerHands[i] = new PokerHand();
+                }
+            }
+            
+            // Initialize the player roles. The user starts as the dealer. 
+            _players = new Dictionary<PlayerRole, int>
```

Comment:
Really, really not a fan of this implementation. Player role's mapped to ints makes it unnecessarily confusing. 

Instead, we can just have "_currentDealer", "_currentSmallBlind", and "_currentBigBlind" integers stored outside of the enum / dictionary mapping. 

That saves some complexity, and means we can also pretty easily update the values when a round is over by just using "++" and a simple check to make sure we don't go over the size of the list. We already have to do this "++" anyways, so why the dictionary?

This would require renaming some of the above variables to be more descriptive: "_currentSmallBlindBet" and "_currentSmallBlindPlayer" would solve.

Reply:
Simpler implementation and would allow a much easier randomization of which player starts out as the dealer upon initialization of Texas Hold Em. Good call.